Upgrade guidance updates from VLT-172 (#13327)
hashicorp/vault: website/content/docs/upgrading/index.mdx

back to a previous version of Vault, you should roll back your data store as
well.

Vault upgrades are designed such that large jumps (ie 1.3.10 -> 1.7.x) are
supported. The upgrade notes for each intervening version must be reviewed. The
upgrade notes may describe additional steps or configuration to update before,
during, or after the upgrade.

## Learn

Refer to the [Vault Upgrade Standard Procedure](https://learn.hashicorp.com/tutorials/vault/sop-upgrade)
guide for step-by-step examples of upgrading Vault Enterprise.

## Agent

The Vault Agent is an API client of the Vault Server. Vault APIs are almost

> always backwards compatible. When they are not, this is called oout in the

Reviewer: ```suggestion
always backwards compatible. When they are not, this is called out in the
```

upgrade guide for the new Vault version, and there is a lengthy deprecation
period. The Vault Agent version can lag behind the Vault Server version, though

> we recommend keeping all versions up to date with the most minor Vault version

Reviewer: This is a little confusing, what does keeping up to date with the most minor Vault version mean?

Author: Fair. First of all I'm missing a word - "most recent minor Vault version". Does that make the sentence feel like it has enough value, or would you be interested in further re-tooling?

Author: ```suggestion
we recommend keeping all Vault instances up to date with the most recent minor Vault version
```

- When satisfied with functionality of upgraded secondary instances, upgrade
  the primary instance

~> **Note:** It is not safe to replicate from a newer version of Vault to an
older version. When upgrading replicated clusters, ensure that upstream clusters

> are always on older versions of Vault than downstream clusters.

Reviewer: ```suggestion
this is a test
```

Author: Going to take "this is a test" at face value.
